frontend/src/pages/product-overview/ProductOverview.jsx

```diff
@@ -38,6 +47,28 @@ function ProductOverview() {
       ),
     },
   ];
+
+  const placeBid = async () => {
+    if (enteredPrice < product.startingPrice) {
+      setMessage(
+        "There are higher bids than yours. You could give a second try."
+      );
+      setMessageStyle("error");
+    }
+    try {
+      const bidResponse = await bid({
+        productId: product.id,
+        price: enteredPrice,
+      });
+      setMessage(bidResponse.message);
+      if (bidResponse.successful) setMessageStyle("success");
+      else setMessageStyle("error");
+    } catch (exception) {
+      setMessageStyle("error");
+      setMessage(exception.response.data);
+    }
```

Comment:
Does this work okay when the user loses internet connection and attempts to bid?